src/app/components/DateRangeControls.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import moment from 'moment';
+import { DateType } from '../api/Api';
+import {
+    Button,
+    Glyphicon
+} from 'react-bootstrap';
+import CalendarButton from './CalendarButton/CalendarButton';
+
+export interface DateRangeControlProps {
+    onNext: () => void;
```

Comment:
well played much better than handling internally.  Let the calling component deal with it.